### ISSUE TITLE
New ViewChangedEvent

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bradrydzewski/gwt-cal#166]
```
Description of the new feature:

Generation of view change event when calendar view is changed in any  way that 
effects the dates/times displayed.

Use Case:

The view mode (day/week/month...) is changed or the date to be viewed is 
changed. The calendar publishes a new ViewChangedEvent which would contain 
attributes indicating the date range that is currently being displayed, the 
view mode.

This would be useful so that logic is not needed to calculate what dates are 
being displayed (calls to MonthsViewDateUtils for example). Business code would 
simply use the data in the event to indicate any appointments that needed to be 
fetched from an appointment service (db/cache/webservice). 

In short this reduces even further the date/time logic required by users code.

Assumptions:
1.
2.
3.

Exceptions:
1.
2.
3.

Code / psuedo-code:

```

Original issue reported on code.google.com by `sweb...@gmail.com` on 10 Aug 2012 at 9:05
